ytypes/leafref.go

```diff
@@ -143,23 +142,27 @@ func leafRefToGNMIPath(root *util.NodeInfo, path string, pathQueryNode *util.Pat
 
 			gp.Key = make(map[string]string)
 			ns, err := dataNodesAtPath(root, pathNoKeysToGNMIPath(v), pathQueryNode)
-			var j []byte
+			var j string
 			switch len(ns) {
 			case 0:
 			case 1:
-				actualValue := ns[0]
 				if err != nil {
 					return nil, err
 				}
-				j, err = json.Marshal(actualValue)
-				if err != nil {
-					return nil, err
+				rval := reflect.ValueOf(ns[0])
+				if util.IsValuePtr(rval) {
```

Comment:
Is this handling the case where there is a pointer key for an element in a list? I can't remember anywhere that ygen would generate code of that nature - aren't all list keys scalars?

Reply:
dataNodesAtPath returns the data node and not the key, so pointers are returned for scalars

Reply:
So if we ask for `/interfaces/interface[name=Ethernet42]` it returns us `config/name` leaf - rather than the map key itself? Makes sense if so.

Reply:
That would return the `oc.Interface` value. In this function, `dataNodesAtPath` is used to get the key value from a path like `current()/../../key`, and since that simply refers to a leaf, it returns the field value, which is a pointer in the case of a scalar. Am I answering the right question?

Reply:
Ah, makes sense.